packages/kmath/src/vector.ts

```diff
@@ -26,6 +26,10 @@ export function zip<T>(...arrays: ReadonlyArray<T>[]): T[][] {
     return results;
 }
 
+export function map<T, U>(pair: [T, T], f: (a: T, i: number) => U): [U, U] {
```

Comment:
**nit**: Most of the other `vector` functions are `ReadonlyArray<T>`.  This is specifically a 2-item tuple variant.  🤔   I guess if we want to make it [n-ary](https://xlinux.nist.gov/dads/HTML/naryfunc.html#:~:text=(definition),%2C%20binary%20function%2C%20trinary%20function.), we can! Never mind me. :)

Reply:
Yeah, the functions that take a `ReadonlyArray<T>` are a problem, because you can't assign their results to `Coord`. Since we are only working in 2 dimensions, I don't see any need to generalize this to more dimensions. If we do need to handle 3D someday, we can add an overload signature.